packages/frontend/src/pages/Templates/index.tsx

```diff
@@ -1,47 +1,92 @@
-import { Center, Flex, Hide, Image, Text } from '@chakra-ui/react'
-import { Badge, Link } from '@opengovsg/design-system-react'
+import { useQuery } from '@apollo/client'
+import { AbsoluteCenter, Box, Center, Flex, Grid, Text } from '@chakra-ui/react'
+import { Link, Tile } from '@opengovsg/design-system-react'
 
-import templatesPreviewImg from '@/assets/templates-preview.svg'
 import Container from '@/components/Container'
-import NavigationDrawer from '@/components/Layout/NavigationDrawer'
+import PageTitle from '@/components/PageTitle'
+import PrimarySpinner from '@/components/PrimarySpinner'
 import * as URLS from '@/config/urls'
+import { Template } from '@/graphql/__generated__/graphql'
+import { GET_TEMPLATES } from '@/graphql/queries/get-templates'
+import { TemplateIcon } from '@/helpers/flow-templates'
+
+const TEMPLATES_TITLE = 'Templates'
 
 export default function Templates(): JSX.Element {
+  const { data, loading } = useQuery(GET_TEMPLATES)
+  const templates: Template[] = data?.getTemplates
+
+  // Sanity check if our templates file is missing
+  if (!templates || templates.length === 0) {
+    return (
+      <Box position="relative" h="100%">
+        <AbsoluteCenter>
+          <Text>
+            There are no templates now, please contact support@plumber.gov.sg
+          </Text>
+        </AbsoluteCenter>
+      </Box>
+    )
+  }
+
```

Comment:
what is this? this will override your loading spinner

Reply:
previously i stored the data in a DB table but now i won't need this sanity check anymore thanks